### PR TITLE
WIP: irmin: move closeable functors into irmin + cleanups

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by samoht** [comment]
Rebased.

File: src/irmin/content_addressable_intf.ml [hunk 1]
```diff
@@ -65,5 +65,11 @@ module type Sigs = sig
     (** @inline *)
   end
 
-  module Check_closed (M : Maker) : Maker
+  module Check_closed (S : S) : sig
```
Comment (Ngoguey42) [inline src/irmin/content_addressable_intf.ml:68]: `Check_closed` -> `Ensure_open`?

File: src/irmin/store_properties_intf.ml [hunk 2]
```diff
@@ -69,4 +87,9 @@ module type Sigs = sig
     include Clearable
     (** @inline *)
   end
+
+  module type Checkable = sig
```
Comment (Ngoguey42) [inline src/irmin/store_properties_intf.ml:91]: I presume this one can't be merged with `Closeable`?

Comment (samoht) [reply src/irmin/store_properties_intf.ml:91]: Yes this functor just adds a check on top for backends which doesn't support double-closing properly. Maybe I should just name this `Double_closeable`? That sounds a bit odd though...

**Comment by samoht** [comment]
I've tried with `Double_closeable` and `Wrap_close` ; that seems to work a bit better. 

We actually need to distinguish between Double_closeable and as we want to wrap all the user-provided close functions in `Make` and `Make_ext` - the backend which needs modifications are the one using `Of_private` directly (where we assume all the internal store are already protected against double-close). Maybe this should appear in the `{Contents,Node,Commit}.Store` too ; I'll think about this tomorrow.

**Comment by Ngoguey42** [comment]
Closing because mainly be cause code is out of date. Referenced by issue #1348

**Pull request closed without merging**